sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -617,3 +617,31 @@ case class DecodeUsingSerializer[T](child: Expression, tag: ClassTag[T], kryo: B
 
   override def dataType: DataType = ObjectType(tag.runtimeClass)
 }
+
+case class InitializeJavaBean(n: NewInstance, fields: Seq[Expression], setters: Seq[String])
```

Comment:
Seems we can not use `Map` here, as the `fields: Seq[Expression]` needs resolution and transformation.